File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/reference/part-two-typescript/src/http.ts
import * as heatmap from '@grafana/grafana-foundation-sdk/heatmap';
import * as logs from '@grafana/grafana-foundation-sdk/logs';
import * as timeseries from '@grafana/grafana-foundation-sdk/timeseries';
import * as prometheus from '@grafana/grafana-foundation-sdk/prometheus';
import * as units from '@grafana/grafana-foundation-sdk/units';
import { Service } from './catalog';
import {
    heatmapPanel,
    logPanel,
    lokiQuery,
    prometheusDatasourceRef,
    prometheusQuery,
    timeseriesPanel,
} from './common';

export const httpRequestsTimeseries = (service: Service): timeseries.PanelBuilder => {
    return timeseriesPanel()
        .title('HTTP Requests')
        .unit(units.RequestsPerSecond)
        .withTarget(
            prometheusQuery(`rate(http_requests_total{service="${service.name}"}[$__rate_interval])`)
                .legendFormat('{{code}} - {{ method }} {{ path }}')
        )
        .datasource(prometheusDatasourceRef())
    ;
};

export const httpLatenciesHeatmap = (service: Service): heatmap.PanelBuilder => {
    return heatmapPanel()
        .title('HTTP Requests latencies')
        .withTarget(
            prometheusQuery(`sum(increase(http_requests_duration_seconds_bucket{service="${service.name}"}[$__rate_interval])) by (le)`)
                .format(prometheus.PromQueryFormat.Heatmap)
        )
        .datasource(prometheusDatasourceRef())
    ;
};

export const httpLogsPanel = (service: Service): logs.PanelBuilder => {
    return logPanel()
        .title('HTTP Logs')
        .withTarget(
            lokiQuery(`{service="${service.name}", source="http", level=~"$logs_level"} |~ "$logs_filter"`)
        )
    ;
};

File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/reference/part-two-golang/http.go
package main

import (
	"fmt"

	"github.com/grafana/grafana-foundation-sdk/go/heatmap"
	"github.com/grafana/grafana-foundation-sdk/go/logs"
	"github.com/grafana/grafana-foundation-sdk/go/prometheus"
	"github.com/grafana/grafana-foundation-sdk/go/timeseries"
	"github.com/grafana/grafana-foundation-sdk/go/units"
)

func httpRequestsTimeseries(service Service) *timeseries.PanelBuilder {
	return timeseriesPanel().
		Title("HTTP Requests").
		Unit(units.RequestsPerSecond).
		WithTarget(prometheusQuery(fmt.Sprintf("rate(http_requests_total{service=\"%s\"}[$__rate_interval])", service.Name)).
			LegendFormat("{{code}} - {{ method }} {{ path }}"),
		).
		Datasource(prometheusDatasourceRef())
}

func httpLatenciesHeatmap(service Service) *heatmap.PanelBuilder {
	return heatmapPanel().
		Title("HTTP Requests latencies").
		WithTarget(prometheusQuery(fmt.Sprintf("sum(increase(http_requests_duration_seconds_bucket{service=\"%s\"}[$__rate_interval])) by (le)", service.Name)).
			Format(prometheus.PromQueryFormatHeatmap),
		).
		Datasource(prometheusDatasourceRef())
}

func httpLogsPanel(service Service) *logs.PanelBuilder {
	return logPanel().
		Title("HTTP Logs").
		WithTarget(lokiQuery(fmt.Sprintf("{service=\"%s\", source=\"http\", level=~\"$logs_level\"} |~ \"$logs_filter\"", service.Name)))
}
